src/kbmod/reprojection.py

```diff
@@ -156,3 +197,231 @@ def reproject_work_unit(work_unit, common_wcs, frame="original"):
+        # for a given list of obstime indices, collect all the science, variance, and mask images.
+        for indices in unique_obstimes_indices:
+            original_wcs = _validate_original_wcs(work_unit, indices, frame)
+            # get the list of images for each unique obstime
+            images_at_obstime = [images[i] for i in indices]
+
+            # convert each image into a science, variance, or mask "image", i.e. a list of numpy arrays.
+            science_images_at_obstime = [this_image.get_science().image for this_image in images_at_obstime]
+            variance_images_at_obstime = [this_image.get_variance().image for this_image in images_at_obstime]
+            mask_images_at_obstime = [this_image.get_mask().image for this_image in images_at_obstime]
+
+            obstimes = [all_obstimes[i] for i in indices]
+
+            # call `_reproject_images` in parallel.
+            future_reprojections.append(
+                executor.submit(
+                    _reproject_images,
+                    science_images=science_images_at_obstime,
+                    variance_images=variance_images_at_obstime,
+                    mask_images=mask_images_at_obstime,
+                    obstimes=obstimes,
+                    common_wcs=common_wcs,
+                    original_wcs=original_wcs,
+                )
+            )
+
+    # when all the multiprocessing has finished, convert the returned numpy arrays to RawImages.
+    concurrent.futures.wait(future_reprojections, return_when=concurrent.futures.ALL_COMPLETED)
+    time_stamps = []
+    image_list = []
+    for result in future_reprojections:
+        science_add, variance_add, mask_add, time = result.result()
+        science_raw_image = RawImage(img=science_add.astype("float32"), obs_time=time)
+        variance_raw_image = RawImage(img=variance_add.astype("float32"), obs_time=time)
+        mask_raw_image = RawImage(img=mask_add.astype("float32"), obs_time=time)
+
+        psf = _get_first_psf_at_time(work_unit, time)
+        # And then stack the RawImages into a LayeredImage.
+        new_layered_image = LayeredImage(
+            science_raw_image,
+            variance_raw_image,
+            mask_raw_image,
+            psf,
+        )
+
+        # append timestamps and layeredImages to lists
+        time_stamps.append(time)
+        image_list.append(new_layered_image)
+
+    # sort the time_stamps, use the ordering to sort image_list
+    image_list = [image_list[i] for i in np.argsort(time_stamps)]
+    stack = ImageStack(image_list)
+
+    # Add the imageStack to a new WorkUnit and return it.
+    new_wunit = WorkUnit(
+        im_stack=stack,
+        config=work_unit.config,
+        wcs=common_wcs,
+        constituent_images=work_unit.constituent_images,
+        per_image_wcs=work_unit._per_image_wcs,
+        per_image_ebd_wcs=work_unit._per_image_ebd_wcs,
+        per_image_indices=unique_obstimes_indices,
```

Comment:
You should also set `reprojected=True` as part of this new `WorkUnit`

Reply:
(realizing this was something I forgot to do originally haha)

Reply:
Done, I can add this to the returned WorkUnit in `_reproject_work_unit` as well.